instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -220,7 +231,8 @@ def process_request(self, request):
 
         request.META[self._environ_activation_key] = activation
         request.META[self._environ_span_key] = span
-        request.META[self._environ_token] = token
+        if token:
+            request.META[self._environ_token] = token
```

Comment:
since this is conditional now, do we need to update places where this is being read to make sure everthing continues to work?

Reply:
Yes Without this check, It was raising an exception while detaching the token from request object

Reply:
I mean earlier we were always setting this value in the dict even if it was `None` but now we won't be setting it at all in some cases. This can result in lookup error if code in other places assumes the key will always be present so we should make sure this doesn't break anything else.

Reply:
Added a check before detaching the token